codepropertygraph/src/main/scala/io/shiftleft/passes/CpgPass.scala

```diff
@@ -12,6 +12,24 @@ import scala.annotation.nowarn
 import scala.concurrent.ExecutionContext
 import scala.concurrent.duration.DurationLong
 
+trait MaybeExecutionContext[T] {
```

Comment:
This typeclass in needed only to be able to constraint type parameter `T` of `CpgPass`. The alternative would be to use unconstrained `T` - it would be less readable but with some comments it may be also fine